PRINCIPLES.md

```diff
@@ -0,0 +1,254 @@
+
+Versions should be uniquely named. This need not be a semantically meanigful name. It is sufficient that each new version is attributed a name that identifies it uniquely. Once a new version has been created, it should be immutable. By this we mena that it should be impossible to modify the relationship between a version's unique name and its value of the Desired State. 
+
+All versions, except to very first,  should also have reference a predecessor or parent, which is another uniquely named version. This enables us to retain a history of the changes.
+
+<!--
+#### Why is it necessary to have versions be immutable and retained indefinitely?
+
+
+
+- only verb is to create a new version.
+This - Only uniquely named new versions of the system configuration can be created and all configuration info is retained (forward only)
+- rollback
+- Audit
+The only exception being in the case of sensitive information leaks due to error.
+  (I would agrue that key rotation is always better option in this case, but for PII and proprietary info this isn't possible).
+- changes as transactions
+
+#### What are the responsibilities of a State store?
+- In addition, state stores must preserve metadata information regarding the identity of agents that create new versions.
+The Desired State of the new version, Who created it, when it was created and ideally, also _why_ it was created. - What Why, who when. 
+
+- Even though we say git, it's actually only true when configured in a very particular way.
+
+#### State stores as coordination points/ collab via common data interface
+- Collab point. 
+- Business rules and security rules about the acceptable state of the system shoudl be checked against any new proposed version of the desired configuration before such changes are accepted as a new version. (Gates before version)
+- where human process is integrated
+- the place where collaboration can be synchronised.
+-->
+
+### 3. Continuous state reconciliation
+
+<div style="background: #E3F2FD; border-left: 3px solid #0D47A1; padding:10px; margin-bottom:5px; font-style:italic">
+Software agents continuously, and automatically, compare a system's <em>Actual State</em> to its <em>Desired State</em>.
+If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
```

Comment:
Should a GitOps compliant system ALWAYS reconcile on identified drift or are other mechanisms possible to "close the loop" (alerting or otherwise?) I think this may be a barrier to a lot of organizations at some initial degrees of capability, although I agree this _should_ be the desired behavior. I guess an automated action could be an alert which triggers a human-lead attempt to reconciliation...

Reply:
"Manually Closing the Loop". Definitely a FAQ point for the website.

Reply:
I think desired state should also include the policy for when reconciliation is allowed to happen.

` I desire changes to Y to only occur during change windows, rollback on A, remediate on B and alert on C`
`I desire X to be Y`